lib/service/servicecfg/config.go

```diff
@@ -594,6 +594,9 @@ func ApplyDefaults(cfg *Config) {
 	cfg.MaxRetryPeriod = defaults.MaxWatcherBackoff
 	cfg.Testing.ConnectFailureC = make(chan time.Duration, 1)
 	cfg.CircuitBreakerConfig = breaker.DefaultBreakerConfig(cfg.Clock)
+
+	// Debug service defaults.
+	cfg.DebugService.Enabled = true
```

Comment:
This introduced a regression. I believe it's now impossible to turn off the debug service through the config.

Connect My Computer turns off the debug service through the config to work around the problem with the path for the debug service socket being too long (see #43250 and #43267). Starting in v16.1.8, despite `debug_service.enabled` being set to `false`, the agent attempts to start the debug service which causes the error shown in #43250.

Reply:
Opened a PR to solve this (#46032). Thanks for reporting it 👍